frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/VolumeClaimTemplateForm.tsx

```diff
@@ -0,0 +1,213 @@
+    setVolumeMode(event.currentTarget.value);
+  };
+
+  const handleStorageClass = (updatedStorageClass): void => {
+    const provisioner: string = updatedStorageClass?.provisioner || '';
+    // if the provisioner is unknown or no storage class selected, user should be able to set any access mode
+    const modes = provisionerAccessModeMapping[provisioner]
+      ? provisionerAccessModeMapping[provisioner]
+      : getAccessModeForProvisioner(provisioner);
+    // setting message to display for various modes when a storage class of a know provisioner is selected
+    const displayMessage =
+      provisionerAccessModeMapping[provisioner] || isCephProvisioner(provisioner)
+        ? t('pipelines-plugin~Access mode is set by storage class and cannot be changed')
+        : t('pipelines-plugin~Permissions to the mounted drive');
+    setAccessMode('ReadWriteOnce');
+    setAccessModeHelp(displayMessage);
+    // setting accessMode to default with the change to Storage Class selection
+    setAllowedAccessModes(modes);
+    setStorageClass(updatedStorageClass?.metadata?.name);
+    setStorageProvisioner(provisioner);
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      setVolumeMode('Filesystem');
+    }
+  };
+
+  const handleRequestSizeChange = (size: RequestSize): void => {
+    const { value, unit } = size;
+    setRequestSizeValue(value);
+    setRequestSizeUnit(unit);
+  };
+
+  React.useEffect(() => {
+    setRequestSizeError(null);
+    const volumeClaimTemplate = {
+      spec: {
+        accessModes: [accessMode],
+        storageClassName: storageClass,
+        volumeMode,
+        resources: {
+          requests: { storage: `${requestSizeValue}${requestSizeUnit}` },
+        },
+      },
+    };
+    if (!requestSizeValue || parseInt(requestSizeValue, 10) < 1) {
+      volumeClaimTemplate.spec.resources.requests.storage = null;
+      setRequestSizeError(t('pipelines-plugin~Size must be an integer greater than 0.'));
+    }
+
+    setFieldValue(name, volumeClaimTemplate);
+    setFieldTouched(name);
+  }, [
+    volumeMode,
+    accessMode,
+    requestSizeValue,
+    requestSizeUnit,
+    storageClass,
+    name,
+    setFieldTouched,
+    setFieldValue,
+    t,
+  ]);
+
+  const helpText =
+    requestSizeValue && requestSizeUnit
+      ? t(
+          'pipelines-plugin~This will create a PersistentVolumeClaim with a size of {{requestSizeValue}} {{requestSizeUnit}}',
+          { requestSizeValue, requestSizeUnit },
+        )
+      : t('pipelines-plugin~This will create a PersistentVolumeClaim');
+
+  return (
+    <FormGroup fieldId={name}>
+      {errors[name] && <Alert isInline variant="danger" title={t('pipelines-plugin~Required')} />}
+      <p className="help-block">{helpText}</p>
+      <ExpandCollapse
+        textExpanded={t('pipelines-plugin~Hide VolumeClaimTemplate options')}
+        textCollapsed={t('pipelines-plugin~Show VolumeClaimTemplate options')}
+      >
+        <p>
```

Comment:
You shouldn't use `p` tags as a means to just force content separation. It's not the intended use of the paragraph tag. If you elements were inline, then it could possibly be used, but still not in the case of a form.

Recommend you just change them to be `div` and add a className of `opp-VolumeClaimTemplateForm--element` or something along that lines. Create a bottom margin/padding of 8px or 4px or whatever is needed to make it similar to the design.